unittests/test-contracts/action_results/action_results.cpp

```diff
@@ -11,5 +13,10 @@ class [[eosio::contract]] action_results : public contract {
          return 10;
       }
 
+      [[eosio::action]]
+      vector<char> retoverlim() {
+         return vector<char>(512, '0');
```

Comment:
I'd like to see tests for:
- exactly at the limit
- exactly one byte over the limit
- The maximum allowed limit

This may require calling `set_action_return_value` directly.